apps/src/lib/kits/maker/boards/microBit/MicroBitBoard.js

```diff
@@ -29,9 +39,7 @@ export default class MicroBitBoard extends EventEmitter {
     /** @private {Object} Map of component controllers */
     this.prewiredComponents_ = null;
 
-    this.chromeOS = isChromeOS();
-
-    let portType = serialPortType(true);
+    const portType = isWebSerialPort(port) ? navigator.serial : SerialPort;
```

Comment:
Interesting thing here . . . `navigator.serial` and `SerialPort` are not exactly interchangable, so we might want to re-examine what we are using `portType` for here. It looks like we pass it to `MBFirmataWrapper` and then `MBFirmataClient`. Then we use it in the `connect` and `setSerialPort` functions. However, we also over-write `setSerialPort` in this PR. So I think we're in a bit of limbo surrounding whether we actually need this `portType` at all. Happy to talk through some options here!

Reply:
(FWIW - I don't think this is a ship-blocker, but we might want to track this in case we want to clean up later.)

Reply:
Thanks! Could you explain the difference between `/* global SerialPort */ // Maybe provided by the Code.org Browser` used on line 2 in `portScanning.js` versus `require('serialport')` used on line 59 in `MBFirmataClient.js`?

Reply:
Just including for documention @epeach 's response from Slack: If we peek at [this file in the Maker App repositor](https://github.com/code-dot-org/browser/blob/main/src/preload.js)y, this is where we are setting the global SerialPort (line 21). We set it a variable instantiated on line 11, which is require('serialport') . So at the end of the day, global SerialPort  and require('serialport`) are giving us the same library. However, because they are pulled from different projects, they might be different versions! But they are both what I call "Node SerialPort".
Ideally, MBFirmataClient.js would simply reference the global SerialPort . However, because we technically don't own that file, we can't assume that everyone else has SerialPort as a global. If you check out my stagnant PR against the microbit foundation, you can see that [one of the things I did](https://github.com/microbit-foundation/microbit-firmata/pull/3/commits/e6501112210d8fb84f075a3df8d6db4ca4d093ef) was change their version of serialPort to be overridden, so that we could pass our global SerialPort to the FirmataClient (and thus ensure we have the correct version) and still allow MBFirmataClient to be re-usable to other organizations.